implement subtracting from dates

frectonz opened the pull request:
I am not sure about the performance of this thing but i have implemented
- subtracting `days`
- subtracting `weeks`
- subtracting `months`
- subtracting `years`

![screenshot](https://user-images.githubusercontent.com/53809656/204468176-83924147-2b2c-48c7-88bd-0b35402187de.png)

printfn commented:
Hi, thanks for your PR! Subtracting days and weeks seems to work well, but I noticed some issues with months and years.

```
> @2022-03-01 - 1 month
Saturday, 29 January 2022
```

This should probably be returning February 1 instead of January 29.

There's also a more general issue with months, which is how to deal with overflow behavior. For example, what's 1 month before 2022-03-31? I think it'd be good to return an error in that case, rather than returning say 2022-02-28 or 2022-03-01.

Similarly, this is also a bit problematic I think:

```
> @2020-02-29 - 1 year
Thursday, 28 February 2019 (should return an error I think)
> @2020-02-28 - 1 year
Wednesday, 27 February 2019 (wrong result, it should be 2019-02-28)
```

frectonz commented:
Fixed the edge cases and added tests for them. I also created an informative error that is returned when the day found by subtraction is not equal to the the input day.

![screenshot_001](https://user-images.githubusercontent.com/53809656/204526707-90e20e12-c69f-488a-babd-dc0e1f608b19.png)

printfn commented:
Nice, that looks great! I fixed another couple of small bugs around leap years, but I think it's now good to go.

mahtar commented:
It looks like addition currently works only for days (e.g. @2025-08-16 + 21 days).
Would it be possible to extend addition to months, years as well?
